image.py

```diff
@@ -40,8 +46,21 @@ def upload(self, image_name, bucket, object_name=None):
                 print(f"{image_name} has been uploaded to {BUCKET_NAME}")
             except S3UploadFailedError:
                 raise
-        # TODO return s3 url
 
-    # TODO generate small link from the real AWS image link
-    def link_image(self):
-        pass
+    def generate_link(self, method_param, method='get_object'):
+        self.s3 = self.session.client('s3')
+        if self.s3 is not None:
+            try:
+                url = self.s3.generate_presigned_url(
+                    ClientMethod=method,
+                    Params=method_param,
+                    ExpiresIn=3600
+                )
+                logger.info(f"Got pre-signed URL: {url}")
```

Comment:
I have a security issue here saying
<Error>
<Code>InvalidRequest</Code>
<Message>The authorization mechanism you have provided is not supported. Please use AWS4-HMAC-SHA256.</Message>
<RequestId>69N01KQ1ZXAZS8XA</RequestId>
<HostId>uHQKbXCaFhnKG13A7sUL8Mb6zkLU2cwlNfCIjx1RDuEpKrck0XPxMg7x/zK3k2bKnz9xf3ii8LQ=</HostId>
</Error>

Reply:
I think what this means is you need to generate a token before you can use the bucket using either the SECRET_KEY or the ACCESS_KEY